hazelcast/src/main/java/com/hazelcast/topic/impl/TopicService.java

```diff
@@ -145,7 +147,14 @@ public void incrementReceivedMessages(String topicName) {
 
     public void publishEvent(String name, TopicEvent event) {
```

Comment:
You can put [here](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/topic/impl/TopicProxySupport.java#L40) a field like `private final TopicConfig config` and can pass `config.multiThreadingEnabled ` to [this method](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/topic/impl/TopicProxySupport.java#L94)